### PR TITLE
Add recipe for tinyarray

### PR DESCRIPTION
**Pull request by basnijholt**

### CODE REVIEW

File: recipes/tinyarray/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - nose
+    - toolchain
+  run:
+    - python
+    - nose
```
Comment (jakirkham) [inline recipes/tinyarray/meta.yaml]: We need `nose` at `build` and `run`?

Comment (basnijholt) [inline recipes/tinyarray/meta.yaml]: You are right, it only needs it for building.

**Comment by patricksnape** [comment]
It looks like this *should* build on Windows, the project is just using some Python functionality that doesn't exist on Windows in the `setup.py` (namely `os.path.samefile`). It looks like this is a custom method for getting the version from git - have you thought about using [versioneer](https://github.com/warner/python-versioneer) instead?

**Comment by patricksnape** [comment]
Add the `<algorithm>` header to get `std::max`

**Comment by basnijholt** [comment]
I'm not really at home with C, so could you maybe help me by giving me the line I can copy paste into [one of these files](https://gitlab.kwant-project.org/basnijholt/tinyarray/tree/master/src)?

**Comment by basnijholt** [comment]
Can someone help me on how I can run [this Python file](https://gitlab.kwant-project.org/basnijholt/tinyarray/blob/master/test_tinyarray.py) as test in `meta.yaml`?

The file is in `MANIFEST.in`, but I don't know where it is coppied.

File: recipes/tinyarray/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+build:
+  # skip: true  # [win]
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - nose
```
Comment (patricksnape) [inline recipes/tinyarray/meta.yaml]: This isn't needed for building.

File: recipes/tinyarray/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - toolchain
+  run:
+    - python
+
+test:
```
Comment (patricksnape) [inline recipes/tinyarray/meta.yaml]: Add:

```
requires:
  - nose
source_files:
  - test_tinyarray.py
commands:
  - $PYTHON test_tinyarray.py     # [unix]
  - %PYTHON% test_tinyarray.py  # [win]
```

File: recipes/tinyarray/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  requires:
+    - nose
+  source_files:
+    - test_tinyarray.py
+  commands:
+    - $PYTHON test_tinyarray.py   # [unix]
```
Comment (patricksnape) [inline recipes/tinyarray/meta.yaml]: OK apparently Jinja doesn't like that, so just merge those two lines into a single one `python test_tinyarray.py`  - sorry!

**Comment by basnijholt** [comment]
Why does this `Error: in section 'test': unknown key 'source_files'` happen?

Outdated `conda-build`?

**Comment by msarahan** [comment]
source_files was added in 2.0.  It will not work with earlier conda-build
versions.

**Comment by basnijholt** [comment]
How can I use 2.0?

**Comment by patricksnape** [comment]
You can't until we fix a `conda-build-all` bug.

**Comment by basnijholt** [comment]
Let me comment out the tests untill then.

**Comment by patricksnape** [comment]
The `setup.py` requires nose for setup but I don't think it should.

**Comment by patricksnape** [comment]
Well, that's not true, if you want to use `setup.py test` - re-add nose to the `build` requirements if you don't want to change the `setup.py` - apologies!

**Comment by basnijholt** [comment]
I opened a [pull request](https://github.com/conda-forge/conda-forge-build-setup-feedstock/pull/32) to fix it (shamelessly copied [this](https://github.com/conda-forge/conda-forge-build-setup-feedstock/issues/28#issuecomment-243856570).

File: recipes/tinyarray/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+{% set name = "tinyarray" %}
+{% set version = "1.1.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: v{{ version }}.tar.gz
+  url: https://gitlab.kwant-project.org/basnijholt/tinyarray/repository/archive.tar.gz?ref=v{{ version }}
```
Comment (patricksnape) [inline recipes/tinyarray/meta.yaml]: So is this a tag or a branch? Is this the main repository?

Comment (basnijholt) [inline recipes/tinyarray/meta.yaml]: This is not the main repository, I've opened a [pull request](https://gitlab.kwant-project.org/kwant/tinyarray/merge_requests/6).

**Pull request merged**